File: 1-100/twoSum/TwoSum.java
import java.util.HashMap;

public class TwoSum {
    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<>();
//        int[] result = {0, 0};
        for (int i = 0; i <= nums.length; i++) {
            if (hm.containsKey(nums[i])) {
                if (nums[i] * 2 == target) {
                    int result[] = {hm.get(nums[i]),i};
                    return result;
                }
            } else if (hm.containsKey(target - nums[i])) {
                int result[] = {hm.get(target - nums[i]),i};
                return result;
            } else {
                hm.put(nums[i], i);
            }
        }
        return null;
    }


    public static void main(String[] args) {
        int[] result = twoSum(new int[]{3,3}, 6);
        for (int i : result) {
            System.out.println(i);
        }
    }
}

File: 1-100/twoSum/twoSum.cpp
#include<iostream>
#include<vector>
#include <unordered_map>
using namespace std;
vector<int> twoSum(vector<int>& nums, int target);
int main()
{
    vector<int> nums ={2, 7, 11, 15};
    int target=9;
    vector<int> secend;
    secend=twoSum(nums, target);
    cout<<secend[0]<<' '<<secend[1]<<endl;

}
vector<int> twoSum(vector<int>& nums, int target){
    unordered_map<int,int>mp;
    for (int i = 0; i < nums.size(); ++i) {
        int fg =target-nums[i];
        if (mp.count(fg)){
            return {mp[fg],i};

        }
        else{
            mp[nums[i]]=i;
        }
    }
    return {};
}
